addons/mail/models/mail_thread.py

```diff
@@ -2371,8 +2371,7 @@ def _notify_prepare_template_context(self, message, msg_vals, model_description=
         else:
             if add_sign:
                 signature = "<p>-- <br/>%s</p>" % author.name
-
-        company = self.company_id.sudo() if self and 'company_id' in self else user.company_id
+        company = (self.company_id.sudo() or self.env.company) if self and 'company_id' in self else user.company_id
```

Comment:
As environment company should always be set as the one you are logged in, makes sense. It could be simplified with
```
company = self.company_id.sudo() if self and 'company_id' in self and self.company_id else self.env.company
```